### ISSUE TITLE
path routing with websockets not working 

### ISSUE THREAD

**Issue by mwaaas** [containous/traefik#920]
**How to reproduce** 
- docker-compose up ( with the docker-compose file below)
- run this command `wscat -c ws://localhost:80/echo/`
am getting `error: Error: unexpected server response (200)`

```yml
traefik:
  image: containous/traefik
  command: --web --docker --docker.domain=docker.localhost --logLevel=DEBUG
  ports:
    - "80:80"
    - "8080:8080"
  volumes:
    - /var/run/docker.sock:/var/run/docker.sock
    - /dev/null:/traefik.toml

whoami1:
  image: emilevauge/whoami
  labels:
    - "traefik.frontend.rule=PathPrefix:/echod/"
    - "traefik.backend=whoami1"
    - "traefik.port=80"
    - "traefik.weight=5"

whoami2:
  image: emilevauge/whoami
  labels:
    - "traefik.frontend.rule=PathPrefix:/echo/"
    - "traefik.backend=whoami2"
    - "traefik.port=80"
    - "traefik.weight=5"
```

**Comment by emilevauge** [comment]
@mwaaas as a workaround, it works replacing `/echo/` by `/echo`.
Curiously, it works with trailing slash without websocket... Needs investigating on this.

**Comment by mwaaas** [comment]
@emilevauge  you are right removed the trailing slash in the docker compose and it worked. I wish was good in golang to contribute this.

**Comment by wpsmithtwc** [comment]
#762

**Comment by ldez** [comment]
Closed by #1638, in the version 1.3.0-rc3.

**Comment by navinSing** [comment]
So what is the fix if we have a [file] and path in the frontend.. 
#1951 
I am able to connect without traefik, but not otherwise
